### PR TITLE
Qt5

### PR DESCRIPTION
**Pull request by mrosanes**
This PR adds support for Qt5 in sardana (and drops support for Qt<4.6)

It fixes most of the items required for #1006 (except those related to trend support under Qt5, wich need replacement of the qwt5 dependency)

### CODE REVIEW

File: src/sardana/taurus/qt/qtgui/extra_sardana/controllertree.py [hunk 1]
```diff
@@ -122,13 +122,13 @@ class ControllerBaseModel(TaurusBaseModel):
...
+    
     def setDataSource(self, pool):
         if self._data_src is not None:
-            Qt.QObject.disconnect(self._data_src, Qt.SIGNAL(
-                'controllerClassesUpdated'), self.controllerClassesUpdated)
+            self._data_src.controllerClassesUpdated.disconnect(self.controllerClassesUpdated)
         if pool is not None:
-            Qt.QObject.connect(pool, Qt.SIGNAL(
-                'controllerClassesUpdated'), self.controllerClassesUpdated)
+            self.pool.controllerClassesUpdated.connect(self.controllerClassesUpdated)
```
Comment (mrosanes) [inline src/sardana/taurus/qt/qtgui/extra_sardana/controllertree.py]: Instead of 'self.pool...' I think only 'pool...' should be used

**Comment by cmft** [comment]
Hi @MikeFalowski  and @teresanunez ,
I will try to start with `sardana.taurus.qt.qtgui.extra_pool.motor` module :)

**Comment by cmft** [comment]
I will continue with the `extra_macroexecutor.sequenceeditor`  modules

**Comment by teresanunez** [comment]
@mrosanes, @cmft, I take extra_pool/poolmotor

**Comment by teresanunez** [comment]
Hi @cmft, yes, I can do it, but if I do not manage today it will have to wait until next week, sorry.

**Comment by teresanunez** [comment]
I tested extra_hkl and it works.

**Comment by cpascual** [comment]
I've done a full code-review and added a few changes. For the rest, it all seems ok to me.

**Comment by reszelaz** [comment]
I've just done the manual tests (from the release how-to) on on this branch using `QT_API=pyqt5`.
Everything worked fine. I just took the following notes:

* In spock, the following WARNING messages are logged in some occassions. But these are especially annoying when a new profile is created cause they hide an important message to the user if he would like to create a new profile or not:
```console
Profile 'demo2' does not exist. Do you want to create one now ([y]/n)?
MainThread     WARNING  2019-01-28 15:44:17,434 TaurusRootLogger: Qtdefault None.None[0]: QSettings::value: Empty key passed
MainThread     WARNING  2019-01-28 15:44:17,440 TaurusRootLogger: Qtdefault None.None[0]: QSettings::value: Empty key passed
MainThread     WARNING  2019-01-28 15:44:17,460 TaurusRootLogger: Qtdefault None.None[0]: Icon theme "Tango" not found.
```
I wonder if these could be eliminated?

* In `expconf`, the "online plot" button is disabled. I assume that this has to do with taurus dropping support to qwt when Qt5 is in use. So, I could not test the rest of the stuff related to the online plotting. Basically, for the moment, Sardana won't have online scan plotting when Qt5 is in use.

* In `showscan`, the experimental scan results are not plotted, but showed in a table (but I think this was already like this).

* In `macroexecutor`, when used in a complete Taurus GUI, I observed a GUI hung, when running a dscan from the "spock yellow line". This happened only once and I can not reproduce it.

I will continue with the `QT_API=pyqt4`

**Comment by cpascual** [comment]
> * In spock, the following WARNING messages are logged in some occassions(...) I wonder if these could be eliminated?

I propose to treat this as an independent issue. Note: I think the ones complaining about empty keys are related to the args passed to TaurusApplication. They can be reproduced with: `from PyQt5 import Qt; app = Qt.QApplication([])`.

> * In `expconf`, the "online plot" button is disabled. I assume that this has to do with taurus dropping support to qwt when Qt5 is in use. So, I could not test the rest of the stuff related to the online plotting. Basically, for the moment, Sardana won't have online scan plotting when Qt5 is in use.

Exactly

> * In `showscan`, the experimental scan results are not plotted, but showed in a table (but I think this was already like this).

This is related to an adaptation of the Nexus widget in taurus for Qt5. I think that if you install `taurus_pyqtgraph`, you will see plots

> I think this is ready to be merged.  I will proceed tomorrow if no one is against.

I agree. Thanks!

One note: maybe now that TEP18 is already merged (and its version bumped), you can make this sardana depend of taurus >=4.5.1

**Comment by reszelaz** [comment]
Taurus requirement was bumped to >= 4.5. However Sardana can be still used with a special branch of Taurus 3 (taurus-org/taurus@3.x-sdn2.5.1). That special branch of Taurus is just maintained for the Sardana clients (Sardana-Taurus widgets, spock, etc.) needs.

Thanks to all for this development! I'm merging it.

**Pull request merged**